psi4/driver/procrouting/proc.py

```diff
@@ -1814,6 +1814,7 @@ def run_dfocc_gradient(name, **kwargs):
         core.set_local_option('DFOCC', 'ORB_OPT', 'TRUE')
 
     core.set_global_option('DERTYPE', 'FIRST')
+    core.set_global_option('QC_MODULE', 'OCC')
```

Comment:
This seems to have been a bad idea as it breaks the default-pathway in the module selection for composite approaches. The default-pathways expect an empty `QC_MODULE` variable. Not sure how else to get module information back to the driver, though.

Reply:
Getting module information back to the driver this way via options is a bad idea. Consider the following example, which I can see myself doing:
```
ocepa_wfn = gradient("ocepa")
dct_wfn = gradient("dct")
fchk(ocepa_wfn, "ocepa")
fchk(dct_wfn, "dct")
```
Your approach cannot possibly determine the module correctly for both of those.

This needs to be determined from the wavefunction somehow. If there isn't a way to do this at present, the first idea I can think of for adding a way is to use the wavefunction name attribute, adding it in OCC if needed.

Reply:
Perhaps one could make a `wfn.set_variable` equivalent for strings? Possibly a standard variable to set the parent module of a `wfn`? Could also be useful for gettings notes and names to a `wfn` object for users, instead of being limited to what the filename allows.

Reply:
if you add QC_MODULE to the OptionsState at the start of the fn, it'll reset when the fn finishes

Reply:
hmm, but that doesn't help with getting the information back to the driver since it resets.

Is `wfn.name()` used by any internal logic, or is it free to modify? Hard to search the code base for it.
At least for `occ` it is inconsistent (aka often missing) anyway. `dfocc` is better.

Something like `name = DF-MP2 [OCC]` could work. Best to avoid round brackets.

Reply:
`wfn.name` should be free to modify. When I gave DCT wavefunctions names, nothing broke.